src/material-experimental/mdc-list/list-base.ts

```diff
@@ -28,47 +34,50 @@ function toggleClass(el: Element, className: string, on: boolean) {
   }
 }
 
-@Directive()
-/** @docs-private */
-export abstract class MatListBase {
-  // @HostBinding is used in the class as it is expected to be extended. Since @Component decorator
-  // metadata is not inherited by child classes, instead the host binding data is defined in a way
-  // that can be inherited.
-  // tslint:disable-next-line:no-host-decorator-in-concrete
-  @HostBinding('class.mdc-list--non-interactive')
-  _isNonInteractive: boolean = false;
-}
-
 @Directive()
 /** @docs-private */
 export abstract class MatListItemBase implements AfterContentInit, OnDestroy, RippleTarget {
+  @HostBinding('tabindex')
+  _tabIndex = -1;
+
   lines: QueryList<ElementRef<Element>>;
 
   rippleConfig: RippleConfig = {};
 
   // TODO(mmalerba): Add @Input for disabling ripple.
   rippleDisabled: boolean;
 
+  _element: HTMLElement;
+
   private _subscriptions = new Subscription();
 
   private _rippleRenderer: RippleRenderer;
 
-  constructor(protected _element: ElementRef, protected _ngZone: NgZone, listBase: MatListBase,
-              platform: Platform) {
-    const el = this._element.nativeElement;
+  protected constructor(private _elementRef: ElementRef<HTMLElement>, protected _ngZone: NgZone,
+                        listBase: MatListBase, platform: Platform) {
+    this._element = this._elementRef.nativeElement;
     this.rippleDisabled = listBase._isNonInteractive;
     if (!listBase._isNonInteractive) {
-      el.classList.add('mat-mdc-list-item-interactive');
+      this._element.classList.add('mat-mdc-list-item-interactive');
     }
     this._rippleRenderer =
-        new RippleRenderer(this, this._ngZone, el, platform);
-    this._rippleRenderer.setupTriggerEvents(el);
+        new RippleRenderer(this, this._ngZone, this._element, platform);
+    this._rippleRenderer.setupTriggerEvents(this._element);
   }
 
   ngAfterContentInit() {
     this._monitorLines();
   }
 
+  ngOnDestroy() {
+    this._subscriptions.unsubscribe();
+    this._rippleRenderer._removeTriggerEvents();
+  }
+
+  _setTabIndexForChildren(value: number) {
```

Comment:
Should we have a description for such method to explain what it does / where it is used? (i.e. in the list foundation).

It would be also good to see how that plays together with markup where developers set the tabindex manually?

Reply:
Yeah, I don't really like this and maybe we can come up with a better way to do it. Its used in their foundation to toggle the focusability of child elements. The idea is to only have the children focusable when the item itself is focusable.

Their whole focus management strategy doesn't really play well with `tabindex` other than `0` and `1`, but at the same time, users probably shouldn't be using other `tabindex` values anyways. @jelbourn do you have any thoughts on this?

My bigger concern is actually the fact that this only disables buttons and anchors (same logic as what MDC has in their component), but it doesn't feel very robust

Reply:
Chatted about this on Slack- we'll go drop this for now and follow-up on this with the MDC team. I think it's unexpected for component users to have any components mutating DOM elements they don't own (in this case user-projected content).